iothub/device/tests/MessageTests.cs

```diff
@@ -165,5 +165,62 @@ public void SettingMessageAsSecurityMessageTest()
             Assert.AreEqual(CommonConstants.SecurityMessageInterfaceId, msg.SystemProperties[MessageSystemPropertyNames.InterfaceId]);
             Assert.IsTrue(msg.IsSecurityMessage);
         }
+
+        [TestMethod]
+        public void CloneWithBodyTest()
+        {
+            using (var original = new Message(Encoding.UTF8.GetBytes("Original copy")))
+            {
+                original.Properties["test1"] = "test_v_1";
+                original.Properties["test2"] = "test_v_2";
+
+                original.ContentEncoding = "gzip";
+                original.ContentType = "text/plain";
+                original.UserId = "JohnDoe";
+
+                using (var clone = original.CloneWithBody(Encoding.UTF8.GetBytes("Cloned version")))
+                { 
+                    Assert.AreEqual("test_v_1", clone.Properties["test1"]);
+                    Assert.AreEqual("test_v_2", clone.Properties["test2"]);
+
+                    Assert.AreEqual("gzip", clone.ContentEncoding);
+                    Assert.AreEqual("text/plain", clone.ContentType);
+                    Assert.AreEqual("JohnDoe", clone.UserId);
+
+                    var clonedContent = default(string);
+
+                    using (var reader = new StreamReader(clone.BodyStream, Encoding.UTF8))
+                    {
+                        clonedContent = reader.ReadToEnd();
+                    }
+                
+                    Assert.AreEqual("Cloned version", clonedContent);
+                }
+            }
+        }
+
+        [TestMethod]
+        public void CloneWithBodyWithNullTest()
```

Comment:
Add SystemProperties as well, so that we cover all cases being impacted.

Reply:
the ' original.ContentType = null;' part goes to SystemProperties I believe

Reply:
ok, I improved the check reading the SystemProperties directly, not through the getter method, which would return "null" even if the value is not in the dictionary